### ISSUE TITLE
defaults override custom-envoronment-variables

### ISSUE THREAD

**Issue by peteroreil** [grimen/python-config2#1]
Hi,

It appears that the `default.yml` is overriding the `custom-environment-variables.yml` defined configuration.

Python Version: **2.7.8**

Project Structure

- ./requirements.txt
- ./.python-version
- ./config/custom-environment-variables.yml
- ./config/default.yml
- ./main.py
- ./env

**main.py**
```
from config2.config import config

import os

print(os.environ['APP']) 
print('port', config.port) 
```

**config/default.yml**
```
port: 3000
```
**config/custom-environment-variables.yml**
```
port: APP
```
When run with `python main.py`  config returns the default (_3000_) as expected
When run with `APP=1111 python main.py` config still returns _3000_ where I expected _1111_ however the `APP` environment variable is returning _1111_.

Is `virtualenv` causing a problem here?

**requirements.txt**
```
attributedict==0.1.8
blessings==1.7
Click==7.0
colour-runner==0.1.1
config2==0.1.4
deepdiff==3.3.0
deepmerge==0.0.4
easypackage==0.1.8
filelock==3.0.9
itsdangerous==0.24
Jinja2==2.10
jsonpickle==1.0
MarkupSafe==1.0
pluggy==0.7.1
py==1.6.0
Pygments==2.2.0
PyYAML==3.13
six==1.11.0
toml==0.10.0
tox==3.5.2
virtualenv==16.0.0
Werkzeug==0.14.1
```

Thanks for your work btw 

**EDIT:**
Just to add to this, defining configuration in another file works fine.
i.e `config/foo.yml` 
and executing `ENV=foo python main.py` will return the correct configuration.

**Comment by grimen** [comment]
@peteroreil Thank you for the bug report, I should look into this issue upcoming days.

**Comment by grimen** [comment]
It looks to me like I completely forgot to cover the `custom-environment-variables` case. :) I need to add tests for that behaviour, and actually by the looks I forgot implementing the mapping logic. I currently don't use ENV overrides in any product code, so explains why I didn't notice.

**Comment by grimen** [comment]
Closed by https://github.com/grimen/python-config2/commit/e7832de6968e3e9351991552842f7768e4aacac6

Will be released as `v0.2.0`
